### PR TITLE
Add created_at column to all the tables. Updated the tests to reflect…

### LINKED ISSUE #71: All tables should have a creation date on them
Only the User table currently has a created_at time. Every other table should have this too.
Knex might have a nice way to add a created_at time to every table. If not, we'll just need to add that as a field to each table in the migration (see User table for details)

### PR DESCRIPTION
**Pull request by Laludia**
closes #71

### CODE REVIEW

File: migrations/20170206142200_setup.js [hunk 1]
```diff
@@ -13,40 +13,48 @@ exports.up = function(knex, Promise) {
...
 			table.integer('user_id').references('user.id').notNullable();
 			table.integer('role_id').references('role.id').notNullable();
+			table.timestamp('created_at').defaultTo(knex.fn.now());
 
 			table.primary(['user_id', 'role_id']);
 		})
 		.createTable('keyword_type', table => {
 			table.increments('id');
 			table.string('name').unique();
+
```
Comment (JDonut) [inline migrations/20170206142200_setup.js]: Remove this blank line

File: test/controllers/user_test.js [hunk 2]
```diff
@@ -137,7 +137,7 @@ describe('User Controller', function() {
 				});
 		});
 
-		it('Cannot update user without valid authentication', function() {
+		it('Cannot update user without valid authentication', function(done) {
```
Comment (JDonut) [inline test/controllers/user_test.js]: This test was actually broken before you added the `done` to it. Nicely done catching that!

File: test/lib/models_test.js [hunk 3]
```diff
@@ -181,109 +179,107 @@ describe('Models', function() {
...
-			GeneTermAnnotation.where({id: testAnnotation.id})
+			return GeneTermAnnotation.where({id: testAnnotation.id})
 				.fetch({withRelated: 'method'})
 				.then(res => {
 					if (!res) throw new Error('No models were returned');
 					let actual = res.toJSON();
-					chai.expect(actual.method).to.deep.equal(expectedKeyword);
-					done();
+					chai.expect(actual.method).to.contain(expectedKeyword);
+
```
Comment (JDonut) [inline test/lib/models_test.js]: Remove this blank line

File: test/lib/models_test.js [hunk 4]
```diff
@@ -181,109 +179,107 @@ describe('Models', function() {
...
-			GeneTermAnnotation.where({id: testAnnotation.id})
+			return GeneTermAnnotation.where({id: testAnnotation.id})
 				.fetch({withRelated: 'keyword'})
 				.then(res => {
 					if (!res) throw new Error('No models were returned');
 					let actual = res.toJSON();
-					chai.expect(actual.keyword).to.deep.equal(expectedKeyword);
-					done();
+					chai.expect(actual.keyword).to.contain(expectedKeyword);
+
```
Comment (JDonut) [inline test/lib/models_test.js]: Remove this blank line

File: test/lib/models_test.js [hunk 5]
```diff
@@ -181,109 +179,107 @@ describe('Models', function() {
...
 
-			GeneTermAnnotation.where({id: testAnnotation.id})
+			return GeneTermAnnotation.where({id: testAnnotation.id})
 				.fetch({withRelated: 'parentData'})
 				.then(res => {
 					if (!res) throw new Error('No models were returned');
 					let actual = res.toJSON();
 					chai.expect(actual.parentData).to.contain(expectedParent);
-					done();
+
```
Comment (JDonut) [inline test/lib/models_test.js]: Blank line

File: test/lib/models_test.js [hunk 6]
```diff
@@ -181,109 +179,107 @@ describe('Models', function() {
...
-			GeneGeneAnnotation.where({id: testAnnotation.id})
+			return GeneGeneAnnotation.where({id: testAnnotation.id})
 				.fetch({withRelated: 'method'})
 				.then(res => {
 					if (!res) throw new Error('No models were returned');
 					let actual = res.toJSON();
-					chai.expect(actual.method).to.deep.equal(expectedKeyword);
-					done();
+					chai.expect(actual.method).to.contain(expectedKeyword);
+					//done();
```
Comment (JDonut) [inline test/lib/models_test.js]: Remove this `//done();` entirely

File: migrations/20170206142200_setup.js [hunk 7]
```diff
@@ -63,15 +71,18 @@ exports.up = function(knex, Promise) {
 			table.integer('method_id').references('keyword.id');
 			table.integer('keyword_id').references('keyword.id');
 			table.integer('evidence_id'); // TODO if we ever have a Locus table, this has a reference to it
+			table.timestamp('created_at').defaultTo(knex.fn.now());
```
Comment (JDonut) [inline migrations/20170206142200_setup.js]: I changed my mind on this one. Annotation has a created_at date already, so it's redundant to add one on all the annotation subtypes as well.

File: migrations/20170206142200_setup.js [hunk 8]
```diff
@@ -63,15 +71,18 @@ exports.up = function(knex, Promise) {
...
 		})
 		.createTable('gene_gene_annotation', table => {
 			table.increments('id');
 			table.integer('locus2_id').notNullable(); // TODO see above note about Locus table reference
 			table.integer('method_id').references('keyword.id');
+			table.timestamp('created_at').defaultTo(knex.fn.now());
```
Comment (JDonut) [inline migrations/20170206142200_setup.js]: Remove this since annotation handles it already

File: migrations/20170206142200_setup.js [hunk 9]
```diff
@@ -63,15 +71,18 @@ exports.up = function(knex, Promise) {
...
 		})
 		.createTable('comment_annotation', table => {
 			table.increments('id');
 			table.string('text').notNullable();
+			table.timestamp('created_at').defaultTo(knex.fn.now());
```
Comment (JDonut) [inline migrations/20170206142200_setup.js]: See above

**Comment by Laludia** [comment]
I believe I updated the changes requested. @JDonut

**Pull request merged**